packages/@o3r/rules-engine/src/engine/operator/operators/date-based.operators.ts

```diff
@@ -16,6 +17,71 @@ export const inRangeDate: Operator<Date, [DateInput, DateInput], DateInput> = {
   validateRhs: isValidDateRange
 };
 
+/**
+ * Validate if the provided operatorFactValues contain 'o3rCurrentTime'
+ * @param value the value
+ */
+function containsO3rCurrentTime(value: unknown): value is Record<string, Facts> {
+  if (typeof value !== 'object' || value === null) {
+    return false;
+  }
+
+  for (const key in value) {
+    if (typeof key !== 'string') {
+      return false;
+    }
+  }
+
+  return Object.prototype.hasOwnProperty.call(value, 'o3rCurrentTime');
+}
+
+/**
+ * Check if the value of the variable is in the next x minutes
+ *
+ * @title is in next minutes
+ *
+ * @returns false for dates before `now` and for dates after `now` + `nextMinutes`, true for dates between `now` and `now` + `nextMinutes`
+ */
+export const dateInNextMinutes: Operator<Date, string | number, DateInput> = {
+  name: 'dateInNextMinutes',
+  evaluator: (leftDateInput, minutes, operatorFactValues) => {
+    if (!operatorFactValues) {
+      throw new Error('No operatorFactValues. Unable to retrieve the current time.');
+    }
+    const currentTimeValue = (operatorFactValues.o3rCurrentTime as Date).getTime();
+    return inRangeDate.evaluator(leftDateInput, [currentTimeValue, currentTimeValue + +minutes * 60000]);
+  },
+  dependencies: ['o3rCurrentTime'],
+  validateLhs: isValidDateInput,
+  validateRhs: numberValidator,
+  validateOperatorFactValues: containsO3rCurrentTime
+};
+
+/**
+ * Check if the value of the variable is not in the next x minutes
+ *
+ * @title is not in next minutes
+ *
+ * @returns false for dates before `now` and for dates between `now` and `now` + `nextMinutes`, true for dates after `now` + `nextMinutes`
+ */
+export const dateNotInNextMinutes: Operator<Date, string | number, DateInput> = {
+  name: 'dateNotInNextMinutes',
+  evaluator: (leftDateInput, minutes, operatorFactValues) => {
+    if (!operatorFactValues) {
+      throw new Error('No operatorFactValues. Unable to retrieve the current time.');
+    }
+    const currentTimeValue = operatorFactValues.o3rCurrentTime as Date;
+    const now = new Date(currentTimeValue);
+    const leftDate = new Date(leftDateInput);
+    const targetDate = new Date(new Date(currentTimeValue).setMinutes(now.getMinutes() + +minutes));
+    return leftDate >= now && leftDate > targetDate;
+  },
+  dependencies: ['o3rCurrentTime'],
+  validateLhs: isValidDateInput,
+  validateRhs: numberValidator,
+  validateOperatorFactValues: containsO3rCurrentTime
```

Comment:
This name is not matching its purpose

Reply:
renamed to `containsO3rCurrentTimeProperty`